src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -70,6 +71,12 @@ public void WriteClient(CodeWriter writer, LowLevelClient client)
                         }
                         else
                         {
+                            var convenienceMethod = client.ConvenienceMethods.FirstOrDefault(convenienceMethod => convenienceMethod.LowLevelClientMethod == clientMethod);
```

Comment:
@annelo-msft , is it important that we write convenience methods right before its client method rather than writing them all at once?

Reply:
I think the answer is yes. See https://github.com/Azure/autorest.csharp/issues/2335#issuecomment-1198164908